guide.md

```diff
@@ -14,7 +14,36 @@ We have this lesson mainly to raise awareness about what factors affect code
 complexity and how to approach modular code design.
 
 
-## Why we should probably rethink this lesson
+## Learning topics
+
+- Know about pure functions (functions without side effects, functions which
+  given same input always return same output).
+- Learn why and how to limit side effects of functions.
+- Discuss why and how to limit side effects of data. Also discuss when
+  mutable data may be preferable.
+- The Zen of Python: https://www.python.org/dev/peps/pep-0020/
+- Discuss why single-purpose functions are often preferred over
+  multi-purpose functions.
```

Comment:
This still seems rather focused on functional programming and the functional level.  Perhaps it could link to #23, #21, #14 could be linked as examples of other things to think about.

Actually, it gives me the idea that all instructor guides could link to these "lesson direction" issues.  Or, we could have a standard label that we apply for them, for every lesson.  Would make it easier to find them.

This doesn't have to be done before merging, it's not critical for the guide and can be just as well added later (and no new issues needed)

Reply:
I agree, still too functional-programming focused. I did not here consider these (valid) issues, only updated the outdated stuff.